server/chat/backend/agent/orchestrator/synthesis.py

```
@@ -0,0 +1,300 @@
+        f"3) rationale: brief reasoning for your decision."
+    )
+
+
+def _fetch_finding_rows(incident_id: str, user_id: str, max_wave: int) -> list:
+    """Return all rca_findings rows for waves 1..max_wave, ordered by wave then start time."""
+    from utils.db.connection_pool import db_pool
+    from utils.auth.stateless_auth import set_rls_context
+
+    try:
+        with db_pool.get_admin_connection() as conn:
+            with conn.cursor() as cur:
```

Ruff (0.15.12)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)